src/axom/core/utilities/BitUtilities.hpp

```diff
@@ -14,9 +14,21 @@
 #ifndef AXOM_BIT_UTILITIES_HPP
 #define AXOM_BIT_UTILITIES_HPP
 
-#include "axom/core/Macros.hpp"  // for AXOM_DEVICE_CODE
+#include "axom/config.hpp"
+#include "axom/core/Macros.hpp"
 #include "axom/core/Types.hpp"
 
+// Check for and setup defines for platform-specific intrinsics
+#if defined(_WIN64) && (_MSC_VER >= 1600)
+  #define _AXOM_CORE_USE_INTRINSICS_MSVC
+  #include <intrin.h>
+#elif defined(__x86_64__) && defined(__GNUC__)
+  #define _AXOM_CORE_USE_INTRINSICS_GCC
+  #include <x86intrin.h>
+#elif defined(__powerpc64__) && (defined(__GNUC__) || defined(__ibmxl__))
+  #define _AXOM_CORE_USE_INTRINSICS_PPC
+#endif
```

Comment:
we will need to add a case for AMD CPUs. please make an issue for this so we don't forget.

Reply:
I created this ticket for AMD: https://github.com/LLNL/axom/issues/759

Reply:
I think clang and gcc usually have the same intrinsics of the format `__builtin_{op}`.

Annoyingly, you can't actually use `__has_builtin` to detect their existence on GCC until like GCC 10.

Reply:
Re: clang -- It's definitely not obvious from the listed conditions, but clang defines `__GNUC__`, so it is captured by the corresponding intel and ppc guards.  

I'll add a comment to the code to clarify this.

Side note:
You can run the following to see the enabled compiler defines and/or flags:
* xlc defines:   ` > xlC -dM -qshowmacros -E <file>`
* clang: 
 Flags: `> clang++   -E - -march=native -###`  
  Defines: `> clang++ -dM -E -x c /dev/null`

Reply:
It took me a while (way too long!) to find the right formulation for the intrinsics on all the platforms/compilers we care about.

Some references:
* https://clang.llvm.org/docs/LanguageExtensions.html#intrinsics-support-within-constant-expressions
* https://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html
* https://stackoverflow.com/questions/11228855/header-files-for-x86-simd-intrinsics
* https://www.ibm.com/docs/en/openxl-c-and-cpp-aix/17.1.0?topic=functions-supported-gcc-non-vector-built-in
* https://docs.microsoft.com/en-us/cpp/intrinsics/intrinsics-available-on-all-architectures?view=msvc-170
* https://en.cppreference.com/w/cpp/header/bit

Reply:
I created an issue to consider renaming our bit utility functions to match the (new) standard functions